projects/hathor-wallet-headless/0001-multisig-create-mint-melt-token.md

```diff
@@ -0,0 +1,327 @@
+
+## Melt a custom token
+
+A single wallet can melt a token using `POST:/wallet/melt-token` endpoint. This operation creates another melt authority by default.
+
+For the MultiSig wallet the endpoint keeps the same command:
+`POST:/wallet/p2sh/tx-proposal/melt-token`
+
+The data fields are:
+* `token`*
+* `amount`*
+* `address`*
+* `change_address`
+* `melt_authority_address`
+* `allow_external_melt_authority_address`
+
+*`*` as required field 
+
+The response scheme for success:
+```ts
+{
+  "success": boolean,
+  "txHex": string
+}
+```
+
+The purpose of the command is only produce the `txHex` not signed and nothing more, it will not push the transaction to the network. The returned `txHex` will be handled by the coordinator service, and later on by the federation service, which is responsible to sign the transaction.
+
+With this specification we cover the possibility to melt tokens, send it to another wallet, create or not another melt authority, and send the authority to another wallet.
+
+## Inspect transaction with wallet metadata
+
+There is an endpoint to [inspect the transaction](https://hathor.gitbook.io/hathor/guides/multisig-wallet/sending-transactions#5.-check-the-multisig-transaction) by decoding the transaction hexadecimal encode in `POST:/wallet/decode`. However, it doesn't bring signature information, nor wallet information with it.
+
+For the MultiSig wallet we have the `inspect` command:
+`POST:/wallet/p2sh/tx-proposal/inspect`
+
+The data field is:
+* txHex
+
+The response scheme for success:
+```ts
+{
+  "success": boolean,
+  "complete": boolean,
+  "tx": {
+    "version": number,
+    "tokens": string[],
+    "inputs": [
+      {
+        tx_id: string,
+        index: number,
+        decoded: {},  // data from the spent ouput
+        signed: boolean,
+        mine: boolean,
+      },
+    ],
+    "outputs": [
+      {
+        value: number,
+        token_data: number,
+        script: string,
+        decoded: {},
+        authorities: number,
+        mine: boolean,
+      }
+    ],
+  },
+  "balance": {
+    "<token-uid>":  {
+      "tokens": { available: number, locked: number ),
+      "authorities": {
+        mint: { available: number, locked: number ),
+        melt: { available: number, locked: number ),
+      },
+    },
+  },
+}
+```
+
+Lets represent the response document as ` $ `, as each element of a list ` [*] `, we have:
+* `$.complete` -- that represents the completeness of signatures required to use the transaction
+* `$.tx.inputs[*].signed` -- that indicates this input has a signature
+* `$.tx.inputs[*].mine` -- that indicates this input belongs to this wallet
+* `$.tx.outputs[*].mine` -- that indicates this output belongs to this wallet
+* `$.balance.tokens` -- that contains the balance of each token in the transaction
+* `$.balance.authorities` -- that contains the balance of mint and melt for each token
```

Comment:
`tokens` and `authorities` are for a single token, and the `balance` is a map where the keys are all tokens in the transaction.

Reply:
I will fix it.

Reply:
Fixed